plugin/ochttp/stats_transport.go

```diff
@@ -79,12 +79,44 @@ func (t statsTransport) CancelRequest(req *http.Request) {
 	}
 }
 
+// StatsHandler is an http.Handler that updates OpenCensus HTTP stats for
+// requests served by a net/http server.
+type StatsHandler struct {
+	Handler http.Handler
+}
+
+// ServeHTTP implements http.Handler and records OpenCensus HTTP stats.
+func (h *StatsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx, _ := tag.New(r.Context(),
+		tag.Upsert(Host, r.URL.Host),
+		tag.Upsert(Path, r.URL.Path),
+		tag.Upsert(Method, r.Method))
+	r = r.WithContext(ctx)
+	track := &tracker{
+		start:  time.Now(),
+		ctx:    ctx,
+		writer: w,
+	}
+	if r.Body == nil {
+		// TODO: Handle cases where ContentLength is not set.
+		track.reqSize = -1
+	} else if r.ContentLength > 0 {
+		track.reqSize = r.ContentLength
+	}
+	stats.Record(ctx, ServerRequestCount.M(1))
+
+	// Perform request.
+	h.Handler.ServeHTTP(track, r)
```

Comment:
We use the convention that a zero-valued http.Handler should mean http.DefaultServeMux, see the existing ochttp.Handler

Reply:
Not a concern post-refactor.